tool/teleport/common/debug.go

```diff
@@ -0,0 +1,294 @@
+	if err != nil {
+		if trace.IsConnectionProblem(trace.ConvertSystemError(err)) {
+			return nil, trace.BadParameter("Unable to reach debug service socket at %q."+
+				"\n\nVerify if you have enough permissions to open the socket and if the path"+
+				" to your data directory (%q) is correct. The command assumes the data"+
+				" directory from your configuration file, you can provide the path to it using the --config flag.", c.socketPath, c.dataDir)
+		}
+
+		return nil, trace.Wrap(err)
+	}
+
+	return resp, nil
+}
+
+func onSetLogLevel(configPath string, level string) error {
+	ctx := context.Background()
+
+	if contains := slices.Contains(logutils.SupportedLevelsText, strings.ToUpper(level)); !contains {
+		return trace.BadParameter("%q log level not supported", level)
+	}
+
+	clt, err := newDebugServiceClient(configPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	setMessage, err := clt.SetLogLevel(ctx, level)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Println(setMessage)
+	return nil
+}
+
+func onGetLogLevel(configPath string) error {
+	ctx := context.Background()
+	clt, err := newDebugServiceClient(configPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	level, err := clt.GetLogLevel(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Printf("Current log level %q\n", level)
+	return nil
+}
+
+// supportedProfiles list of supported pprof profiles that can be collected.
+// This list is composed by runtime/pprof.Profile and http/pprof definitions.
+var supportedProfiles = map[string]struct{}{
+	"allocs":       {},
+	"block":        {},
+	"cmdline":      {},
+	"goroutine":    {},
+	"heap":         {},
+	"mutex":        {},
+	"profile":      {},
+	"threadcreate": {},
+	"trace":        {},
+}
+
+// defaultCollectProfiles defines the default profiles to be collected in case
+// none is provided.
+var defaultCollectProfiles = []string{"goroutine", "heap", "profile"}
+
+func onCollectProfile(configPath string, rawProfiles string, seconds int, out io.Writer) error {
+	ctx := context.Background()
+
+	profiles := defaultCollectProfiles
+	if rawProfiles != "" {
+		profiles = strings.Split(rawProfiles, ",")
+	}
+
+	for _, profile := range profiles {
+		if _, ok := supportedProfiles[profile]; !ok {
+			return trace.BadParameter("%q profile not supported", profile)
+		}
+	}
+
+	clt, err := newDebugServiceClient(configPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var output bytes.Buffer
+	if err := createProfilesArchive(ctx, clt, &output, profiles, seconds); err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Fprint(out, output.String())
+	return nil
+}
+
+// createProfileArchive collects the profiles and generate a compressed tarball
+// file.
+func createProfilesArchive(ctx context.Context, clt *debugServiceClient, buf io.Writer, profiles []string, seconds int) error {
+	fileTime := time.Now()
+	gw := gzip.NewWriter(buf)
+	tw := tar.NewWriter(gw)
+
+	for _, profile := range profiles {
```

Comment:
Wonder if we can print some progress to stderr. When I run the default profile command, I have no idea what it is doing (or if its working at all...)

Reply:
I've added a simple progress bar using [Bubbles](https://github.com/charmbracelet/bubbles) (we already used it for `tsh latency` command). Here is a short preview:

![collect](https://github.com/gravitational/teleport/assets/2467749/b1205435-44ff-4565-aadc-9257d762fdad)

Reply:
I love the idea but I run into a few issues with it
1. If I run the default `profile` command  (with 3 profiles), the progress bar goes to 67% and then to 33%
2. the command fails without a PTY 
```
 $ tsh ssh root@node /go/teleport/e/build/teleport debug -c /opt/teleport/teleport_app.yaml profile --seconds 2 >  test.tar.gz
ERROR: could not open a new TTY: open /dev/tty: no such device or address

ERROR: Process exited with status 1
```
If you add `--tty` flag, it will run but you don't see the progress bar.

What do you say that we leave this out for a future change and just keep it simple for now?